### PR TITLE
PROD-839 Fix 2287 Spammed Discussion Content with documents are removed and unable to recover after unspam

### LINKED ISSUE #2287: Spammed Discussion Content with documents are removed and unable to recover after unspam.
**Describe the bug**
Spammed Discussion Content ( eg. reply ) with documents are removed and unable to recover after unspam.

**To Reproduce**
1. Go to any discussion and create a reply with the document
2. Mark as spam
3. The document gets removed and cannot be recovered even after unspam.

**Expected behavior**
The Unspammed content with the documents should be shown

**Screenshots**
https://prnt.sc/10zttzq

image

**Support ticket links**
https://secure.helpscout.net/conversation/1467715563/133315/

### PR DESCRIPTION
**Pull request by bb-yudhisthir**
Mark related activity as spam if topic/reply marked as spam instead earlier was deleting it which was permanently removing attachment

### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2287.

### How to test the changes in this Pull Request:

1. Go to any discussion and create a reply with the document
2. Mark as spam
3. The document gets removed and cannot be recovered even after unspam.

### Proof Screenshots or Video

https://screencast-o-matic.com/watch/crQZ6rV64fK

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Mark related activity as spam if topic/reply marked as spam

### CODE REVIEW

**Review by dipskakadiya — DISMISSED** [review]

File: src/bp-forums/activity.php [hunk 1]
```diff
@@ -647,6 +647,23 @@ public function topic_update( $topic_id = 0, $post = null ) {
 				$topic_author_id = bbp_get_topic_author_id( $topic_id );
 
 				$this->topic_create( $topic_id, $forum_id, array(), $topic_author_id );
+			} else if( bbp_get_spam_status_id() === $post->post_status ){
+				
+				//Mark related activity as spam if topic marked as spam
```
Comment (jitendrabanjara1991) [inline src/bp-forums/activity.php:652]: // Mark related activity as spam if topic marked as spam.
Same thing on line no 826

**Review by jitendrabanjara1991 — CHANGES_REQUESTED** [review]

File: src/bp-forums/activity.php [hunk 2]
```diff
@@ -647,6 +647,23 @@ public function topic_update( $topic_id = 0, $post = null ) {
 				$topic_author_id = bbp_get_topic_author_id( $topic_id );
 
 				$this->topic_create( $topic_id, $forum_id, array(), $topic_author_id );
+			} else if( bbp_get_spam_status_id() === $post->post_status ){
```
Comment (jitendrabanjara1991) [inline src/bp-forums/activity.php:650]:
```suggestion
			} elseif( bbp_get_spam_status_id() === $post->post_status ) {
```

File: src/bp-forums/activity.php [hunk 3]
```diff
@@ -804,6 +821,23 @@ public function reply_update( $reply_id, $post ) {
 				$reply_author_id = bbp_get_reply_author_id( $reply_id );
 
 				$this->reply_create( $reply_id, $topic_id, $forum_id, array(), $reply_author_id );
+			} else if( bbp_get_spam_status_id() === $post->post_status ){
```
Comment (jitendrabanjara1991) [inline src/bp-forums/activity.php:824]:
```suggestion
			} elseif( bbp_get_spam_status_id() === $post->post_status ) {
```

File: src/bp-forums/activity.php [hunk 4]
```diff
@@ -647,6 +647,23 @@ public function topic_update( $topic_id = 0, $post = null ) {
...
 				$this->topic_create( $topic_id, $forum_id, array(), $topic_author_id );
+			} else if( bbp_get_spam_status_id() === $post->post_status ){
+				
+				// Mark related activity as spam if topic marked as spam.
+				if ( $activity_id = $this->get_activity_id( $topic_id ) ) {
+					
+					$activity = new BP_Activity_Activity( $activity_id );
+					
+					if ( empty( $activity->id ) )
+					return false;
```
Comment (jitendrabanjara1991) [inline src/bp-forums/activity.php:658]:
```suggestion
					if ( empty( $activity->id ) ) {
						return false;
					}
```

File: src/bp-forums/activity.php [hunk 5]
```diff
@@ -804,6 +821,23 @@ public function reply_update( $reply_id, $post ) {
...
 				$this->reply_create( $reply_id, $topic_id, $forum_id, array(), $reply_author_id );
+			} else if( bbp_get_spam_status_id() === $post->post_status ){
+				
+				// Mark related activity as spam if reply marked as spam.
+				if ( $activity_id = $this->get_activity_id( $reply_id ) ) {
+					
+					$activity = new BP_Activity_Activity( $activity_id );
+					
+					if ( empty( $activity->id ) )
+					return false;
```
Comment (jitendrabanjara1991) [inline src/bp-forums/activity.php:832]:
```suggestion
					if ( empty( $activity->id ) ) {
						return false;
					}
```

**Review by jitendrabanjara1991 — CHANGES_REQUESTED** [review]

**Review by jitendrabanjara1991 — APPROVED** [review]

**Pull request merged**